gomock/controller.go

```diff
@@ -80,7 +83,7 @@ func NewController(t TestReporter) *Controller {
 	}
 }
 
-func (ctrl *Controller) RecordCall(receiver interface{}, method string, args ...interface{}) *Call {
+func (ctrl *Controller) RecordCall(receiver interface{}, method string, methodType reflect.Type, args ...interface{}) *Call {
```

Comment:
If backwards compatibility is an issue then it could be maintained by keeping the original method signature and adding a new `RecordCallWithMethodType` (or similar) method which would be used by the generator.

`RecordCall` could simply forward the call to `RecordCallWithMethodType` with a methodType==nil parameter which would force the methodType to be looked up at runtime (like the original implementation).